### PR TITLE
Add warning messages for required EPUB3 metadata

### PR DESCRIPTION
**Pull request by shibukawa**
Subject: <short purpose of this pull request>

### Feature or Bugfix
- Feature

### Purpose
- Detect warning of EPUB validator during Sphinx build

### Detail
- Check the required properties are empty or not

### Sample conf.py

```
epub_contributor = ''
epub_description = ''
epub_uid = ''
epub_publisher = ''
epub_title = ''
epub_author = ''
epub_language = ''
epub_identifier = ''
epub_copyright = ''
epub_scheme = ''
```

### Sample warning messages

```
$ make epub                                                                                                                                                                      [fix/epub_dublincore_validation_error]
Running Sphinx v1.5.3+/00091863
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [epub]: targets for 1 source files that are out of date
updating environment: 1 added, 0 changed, 0 removed
reading sources... [100%] index                                                                                                                                                                                                            
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index                                                                                                                                                                                                             
generating indices... genindex
writing additional pages...
copying static files... done
copying extra files... done
WARNING: conf value "epub_language" (or "language")  should not be empty for EPUB3
WARNING: conf value "epub_uid" should not be empty for EPUB3
WARNING: conf value "epub_title" (or "html_title")  should not be empty for EPUB3
WARNING: conf value "epub_author" should not be empty for EPUB3
WARNING: conf value "epub_contributor" should not be empty for EPUB3
WARNING: conf value "epub_description" should not be empty for EPUB3
WARNING: conf value "epub_publisher" should not be empty for EPUB3
WARNING: conf value "epub_copyright" (or "copyright") should not be empty for EPUB3
WARNING: conf value "epub_identifier" should not be empty for EPUB3
WARNING: conf value "version" should not be empty for EPUB3
```

### Test result of EPUB validator

<img width="1031" alt="2017-02-26 10 40 14" src="https://cloud.githubusercontent.com/assets/564612/23336335/fe9575ca-fc0f-11e6-96e9-d53e8890bc45.png">

### CODE REVIEW

File: sphinx/builders/epub3.py
Comment (tk0miya) [inline sphinx/builders/epub3.py]: Why is this changed? I think "unknown" is not a good default setting. And users would not be noticed if this will be filled by default. I think the description should be filled by user manually.

Note: If changed, we should also update `doc/config.rst`.

**Review by tk0miya — CHANGES_REQUESTED** [review]
Basically LGTM, but I think this is an improvement. so it should move to master branch.
I know this is very small change and might not hurt anyone. But, in my experience, adding warning confuse the less people (and it brings new github issue).

File: sphinx/builders/epub3.py
Comment (shibukawa) [reply sphinx/builders/epub3.py]: Other settings of epub use same default value (``unknown``):

* ``epub_contributor``
* ``epub_author``
* ``epub_publisher``
* ``epub_identifier``
* ``epub_scheme``
* ``epub_uid``

**Comment by shibukawa** [comment]
I migrated from stable to master

File: sphinx/builders/epub3.py
Comment (tk0miya) [reply sphinx/builders/epub3.py]: Hmm... okay, you're right. Let's go with `unknown`.

**Review by tk0miya — APPROVED** [review]

**Comment by tk0miya** [comment]
Thank you for rebasing, but flake8 still warns. Could you check it please?
```
./sphinx/builders/epub3.py:80:96: E501 line too long (102 > 95 characters)
./sphinx/builders/epub3.py:86:96: E501 line too long (101 > 95 characters)
./sphinx/builders/epub3.py:101:96: E501 line too long (103 > 95 characters)
```

**Comment by shibukawa** [comment]
fixed

**Comment by tk0miya** [comment]
Thank you for update. But it seems our master branch fails tests, and it causes this failed.
https://travis-ci.org/sphinx-doc/sphinx/builds/205988293

To prevent spreading the errors, I will merge this after @jfbu fixed the testcase.
(I will also try to check it later (maybe tonight))

**Comment by jfbu** [comment]
@shibukawa @tk0miya sorry about my mistake. After merging #3429 I realized the recently introduced longtable template needed an update, but I forgot to update a test :confounded: 

this is now fixed at 3e3d97f and Travis passes.

**Pull request merged**
